pkg/vm/engine/disttae/txn_table.go

```diff
@@ -729,12 +721,13 @@ func (tbl *txnTable) rangesOnePart(
 		}, uncommittedObjects...)
 	}
 
-	for _, entry := range tbl.writes {
+	// XXX By all means this should by calling ForEachTableWritesStable.
+	tbl.db.txn.ForEachTableWritesIncludingSelf(tbl.db.databaseId, tbl.tableId, func(entry *Entry) {
```

Comment:
That's not how it should be handled here. For the current SQL you just need to see the operation of the previous SQL. This operation may cause this SQL to see its own writes.